src/prefect/artifacts.py

```diff
@@ -173,6 +177,24 @@ async def format(self) -> str:
         return json.dumps(self._sanitize(self.table))
 
 
+class ProgressArtifact(Artifact):
+    progress: float
+    type: Optional[str] = "progress"
+
+    async def format(self) -> float:
+        # Ensure progress is between 0 and 100
+        min_progress = 0.0
+        max_progress = 100.0
+        if self.progress < min_progress or self.progress > max_progress:
```

Comment:
Is it possible to centralize validation handling for `progress` being between 0 and 100?
Note this might be tricky with pydantic changes also happening

Reply:
@zangell44 now that I've merged the pydantic change, what do you suggest?

Reply:
Summarizing offline chat: the correct validation would use a new constrained annotated type https://docs.pydantic.dev/2.7/api/types/#pydantic.types.confloat--__tabbed_1_2. However, this would fail artifact creation if math is incorrect, which might not be the best UX. I defer to you on how best to handle it

Reply:
Discussed with @WillRaphaelson as well -- I think we're good with the current implementation